rust/private/rustc.bzl

```diff
@@ -628,7 +628,12 @@ def construct_arguments(
     rustc_flags.add_all(rust_std_paths, before_each = "-L", format_each = "%s")
     rustc_flags.add_all(rust_flags)
 
-    data_paths = getattr(attr, "data", []) + getattr(attr, "compile_data", [])
+    # Deduplicate data paths due to https://github.com/bazelbuild/bazel/issues/14681
+    data_paths = getattr(attr, "data", [])[:]
+    for extra_data in getattr(attr, "compile_data", []):
+        if extra_data not in data_paths:
+            data_paths.append(extra_data)
```

Comment:
Would it be more efficient to put these into a depset?

Reply:
We need to pass a sequence, not a depset, to the function, but we might as well use depset to do the deduping for us! Done.